tools/cutadapt/main.nf

```diff
@@ -5,44 +5,49 @@ nextflow.enable.dsl=2
 
 // Trimming reusable component
 process cutadapt {
-    publishDir "${params.outdir}/cutadapt",
-        mode: "copy", overwrite: true
+    publishDir "${params.outdir}/${opts.publish_dir}",
+    mode: "copy", 
+    overwrite: true,
+    saveAs: { filename ->
+                    if (opts.publish_results == "none") null
+                    else filename }
     
     container 'luslab/nf-modules-cutadapt:latest'
 
     input:
-        tuple val(sample_id), path(reads)
+        val opts
+        tuple val(meta), path(reads)
 
     output:
-        tuple val(sample_id), path("*.trimmed.fq.gz"), emit: trimmedReads
```

Comment:
can we change channel to fastq? i am trying to keep it as the file type if possible unless it really doesn't make sense